Tests/BotTests/MergeService/MergeServiceTests.swift

```diff
@@ -239,6 +244,12 @@ class MergeServiceTests: XCTestCase {
             },
             assert: {
                 expect($0) == [
+                    // First received new PR with no integration label, so we end up creating the MergeService but destroying it right away since the queue is empty
+                    .created(branch: MergeServiceFixture.defaultTargetBranch),
+                    .state(.stub(status: .starting)),
+                    .state(.stub(status: .idle)),
+                    .destroyed(branch: MergeServiceFixture.defaultTargetBranch),
+                    // Then received PR event about adding integration label, so we end up creating the MergeService again but this time for good
                     .created(branch: MergeServiceFixture.defaultTargetBranch),
                     .state(.stub(status: .starting)),
```

Comment:
Notice the new behavior: with the new implementation that avoids `DispatchService` having to know about the internal logic and impl details of `MergeService`, that means that it always creates a `MergeService` even if that `MergeService` ends up deciding there's nothing to do and gets immediately destroyed… and gets recreated soon after when we receive the `.labeled` event.